go-controller/pkg/allocator/mac/reservation_test.go

```diff
@@ -0,0 +1,75 @@
+package mac_test
+
+import (
+	"net"
+
+	"github.com/ovn-org/ovn-kubernetes/go-controller/pkg/allocator/mac"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("ReservationManager", func() {
+	const owner1 = "namespace1/pod1"
+	const owner2 = "namespace2/pod2"
+
+	var testMgr *mac.ReservationManager
+	var mac1 net.HardwareAddr
+
+	BeforeEach(func() {
+		var err error
+		mac1, err = net.ParseMAC("aa:bb:cc:dd:ee:f1")
+		Expect(err).NotTo(HaveOccurred())
+
+		testMgr = mac.NewManager()
+	})
+
+	Context("reserve", func() {
+		DescribeTable("should not reserve MAC, given",
+			func(owner string, mac net.HardwareAddr) {
+				Expect(testMgr.Reserve(owner, mac)).To(Succeed())
+
+				Expect(testMgr.Reserve(owner1, mac1)).To(Succeed(), "mac should not be reserved")
+			},
+			Entry("zero-values", "", nil),
+			Entry("no owner", "", mac1),
+			Entry("no mac", owner1, nil),
+		)
+		It("should not fail on repeated reservation for the same owner", func() {
```

Comment:
You checked:
1. same owner, same MAC
2. diff owner, same MAC
For wholeness please add mundane happy paths:
3. same owner, diff MAC
4. diff owner, diff MAC

Reply:
I exported (2) to its own tests, and added test for (4)

Reply:
Following offline discution, input validation checks were removed and the releated tests. Input is validated at the caller.